src/wad/image.rs

```diff
@@ -152,11 +186,11 @@ impl Image {
         let (x_start, y_start) = (x_start as usize, y_start as usize);
 
         let source_rows = source.pixels[x_start + y_start * src_pitch..]
-                                .chunks(src_pitch)
-                                .take(copy_height)
-                                .map(|row| &row[..copy_width]);
-        let dest_rows = self.pixels[(x_start as isize + offset[0]) as usize
-                                    + (y_start as isize + offset[1]) as usize * dest_pitch..]
+                              .chunks(src_pitch)
```

Comment:
This seems like a bug: it should be aligned with `source.pixels`, the way it's below. Could you please fix it up?